### PR TITLE
Vite build server

### PR DESCRIPTION
**Pull request by kevinvanrossum**
I've added the Vite build server to the project, so we easily can in the future, run a testing suite to make sure code changes don't break the app.

Since this is a pretty big change in the structure of the app, no more direct source that will be copied via ftp. It will now be build in to a distributable in the dist folder. Since I can't test the deployment flow in the github actions that well. @MichielvanBeers you probably should test if this works, since it can fail and need some fine tuning.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    onSparklyButtonClick,
+    onTeamToggleChange
+} from "./eventListeners.js";
+import {getToggleValue} from "./localStorage.js";
+import {teams} from "./teams.js";
+
+
+let weights = {};
+
+function initializeWeights() {
```
Comment (kevinvanrossum) [inline src/app.js:13]: #6 Is now fixed by only loading in new users in the initialize method

**Comment by MichielvanBeers** [comment]
Very nice @kevinvanrossum!

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -10,6 +10,15 @@ jobs:
     steps:
     - name: 🚚 Get latest code
       uses: actions/checkout@v3
+
+    - name: 📀 Setup Node
+      uses: actions/setup-node@v3
+
+    - name: 🏗️ Install dependencies
+      uses: bahmutov/npm-install@v1`
```
Comment (MichielvanBeers) [inline .github/workflows/deploy.yml:18]: @kevinvanrossum I get a 404 for this NPM install package. Why not just run the command directly?

Comment (MichielvanBeers) [reply .github/workflows/deploy.yml:18]: eg `run: npm install`

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -18,6 +27,7 @@ jobs:
         username: ${{ secrets.FTP_USERNAME }}
         password: ${{ secrets.FTP_PASSWORD }}
         server-dir: public_html/
```
Comment (MichielvanBeers) [inline .github/workflows/deploy.yml]: @kevinvanrossum if you temporarily set the server-dir to /test (or something) and then let it trigger on your branch, we have a testing URL as well, right? I'll set up later to have a testing subdomain like test.whoneedstosharetheboard.xyz

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (MichielvanBeers) [reply .github/workflows/deploy.yml:18]: @kevinvanrossum

**Review by MichielvanBeers — COMMENTED** [review]
Lekker bezig

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (DriesMeerman) [reply .github/workflows/deploy.yml:18]: For which package do you get a 404?

The task should just do npm install but handles container stuff like making sure npm is in the path etc
So you don't get a npm not allowed to run or not found itself.

Comment (MichielvanBeers) [reply .github/workflows/deploy.yml:18]: @DriesMeerman when trying to resolve it from my IDE. Haven't run the pipeline itself. But if @kevinvanrossum runs the pipeline via the suggestion below (of deploying to {url}/test) then we can see if it works and if it is just an error from my side.

**Pull request merged**